kdumpctl

```diff
@@ -1653,6 +1662,173 @@ _should_reset_crashkernel() {
 	[[ $(kdump_get_conf_val auto_reset_crashkernel) != no ]] && systemctl is-enabled kdump &> /dev/null
 }
 
+set_kdump_test_id()
+{
+	local _id=$1
+	local _kmsg="/dev/kmsg"
+
+	if [[ $DEFAULT_DUMP_MODE == "fadump" ]]; then
+		[[ -c "$_kmsg" ]] && echo "$_id" > "$_kmsg"
+	else
+		KDUMP_COMMANDLINE_APPEND+=" $_id "
+		reload >& /dev/null
+	fi
+
+	if [[ "$?" -ne 0 ]]; then
+		derror "Set kdump test id fail."
+		exit 1
+	fi
+}
+
+# $1: success/fail/pending/manual/clear
+# $2: test id
+set_vmcore_creation_status()
+{
+	local _status=$1
+	local _test_id=$2
+	_dir=$(dirname "$VMCORE_CREATION_STATUS")
+
+	[[ -d "$_dir" ]] || mkdir -p "$_dir"
+	[[ -w "$_dir" ]] || chmod +w "$_dir"
+
+	case "$_status" in
+		pending)
+			echo  "$_status $_test_id" > "$VMCORE_CREATION_STATUS"
+			;;
+		success | fail | manual)
+			sed -E -i "s/^\w+/$_status/" "$VMCORE_CREATION_STATUS"
+			;;
+		clear)
+			rm -f "$VMCORE_CREATION_STATUS"
+			;;
+		*)
+			return
+	esac
+	sync -f "$_dir"
+}
+
+fetch_test_status()
+{
+	local _test_id="$1"
+	local _script="$KDUMP_TMPDIR/fetch.sh"
+
+	cat << EOF > "$_script"
```

Comment:
Hi @liutgnu, why do you create a temporary script `$KDUMP_TMPDIR/fetch.sh` first? Can't it run as a function?

Reply:
The script is for iterate the target folder for kexec-dmesg.log and vmcore-dmesg.txt which holding the test_id string. For local iteration, the script can be simplified into a function, but for ssh dump, I didn't find a way to execute a function via ssh, but only create a script file and ask ssh to execute it remotely.Do you have better solutions for that?

Reply:
A function won't work for ssh dump, thanks for enlightening me! It seems there is no need for variable expansion in fetch. You can simply use `cat << \EOF > "$_script"` to avoid escaping `$1` thus to increase code readability. If `fetch.sh` get larger, it's better not to generate on the fly so shellcheck can do some static analysis. 

Btw, why `--` is used in `ssh -i "${OPT[sshkey]}" -o BatchMode=yes "${OPT[_target]}" 'bash -s' -- < "$_script" "${OPT[path]}"`?

Reply:
I will try to get rid of the temporarty script file, as Philipp pointed out, the script file can be hacked on the disk, which may leave security issues. No script may be better. "bash -s -- < script argument" will ask bash to execute the script with the argument passed to the script, seems the same as "bash -s < script argument"...

Reply:
See v2, which I removed the temporary script file to fetch test results.